app/src/androidTest/java/com/github/se/project/ui/profile/ProfileInfoScreenTest.kt

```diff
@@ -0,0 +1,210 @@
+          lastName = "Doe",
+          phoneNumber = "1234567890",
+          role = Role.TUTOR,
+          section = Section.GM,
+          academicLevel = AcademicLevel.MA2,
+          schedule = List(7) { List(12) { 0 } },
+          price = 50)
+
+  private val mockStudentProfile =
+      Profile(
+          uid = "1",
+          googleUid = "1",
+          firstName = "Student",
+          lastName = "Test",
+          phoneNumber = "1234567890",
+          role = Role.STUDENT,
+          section = Section.GM,
+          academicLevel = AcademicLevel.MA2)
+
+  @get:Rule val composeTestRule = createComposeRule()
+
+  @Before
+  fun setUp() {
+    // Mock the dependencies
+    mockLessonRepository = mock(LessonRepository::class.java)
+    mockProfilesRepository = mock(ProfilesRepository::class.java)
+    mockNavigationActions = mock(NavigationActions::class.java)
+
+    // Create a real instance of ListProfilesViewModel
+    mockListProfilesViewModel = ListProfilesViewModel(mockProfilesRepository)
+    mockLessonViewModel = LessonViewModel(mockLessonRepository)
+
+    // Use a Mockito spy to track the actual state of the view model
+    mockListProfilesViewModel = spy(mockListProfilesViewModel)
+
+    // Stub the updateProfile method to simulate a successful update
+    doNothing().`when`(mockProfilesRepository).init(any())
+    whenever(mockProfilesRepository.updateProfile(any(), any(), any())).thenAnswer { invocation ->
+      val onSuccess = invocation.arguments[1] as () -> Unit
+      onSuccess() // Simulate a successful update
+    }
+  }
+
+  @Test
+  fun profileInfoScreen_everythingDisplayedCorrectly() {
+    composeTestRule.setContent {
+      ProfileInfoScreen(
+          navigationActions = mockNavigationActions,
+          listProfilesViewModel = mockListProfilesViewModel,
+          lessonViewModel = mockLessonViewModel)
+    }
+
+    composeTestRule.onNodeWithTag("profileTopBar").assertIsDisplayed()
+    composeTestRule.onNodeWithTag("calendarButton").assertIsDisplayed()
+    composeTestRule.onNodeWithTag("editProfileButton").assertIsDisplayed()
+    composeTestRule.onNodeWithTag("closeButton").assertIsDisplayed()
+  }
+
+  @Test
+  fun tutorProfileInfoScreenDisplaysProfileDetails_whenProfileIsNotNull() {
+    // Mock the currentProfile StateFlow to return the mockProfile
+    val currentProfileFlow = MutableStateFlow<Profile?>(mockTutorProfile)
+
+    // Stubbing the StateFlow
+    doReturn(currentProfileFlow).`when`(mockListProfilesViewModel).currentProfile
+
+    composeTestRule.setContent {
+      ProfileInfoScreen(
+          navigationActions = mockNavigationActions,
+          listProfilesViewModel = mockListProfilesViewModel,
+          lessonViewModel = mockLessonViewModel)
+    }
+
+    // Check if profile details are displayed
+    composeTestRule
+        .onNodeWithTag("profileName")
+        .assertIsDisplayed()
+        .assertTextEquals("${mockTutorProfile.firstName} ${mockTutorProfile.lastName}")
```

Comment:
I did not know it was possible to write with variables inside the assertTextEquals, it's good to know thanks!